pyproject.toml

```diff
@@ -82,7 +82,7 @@ keywords = ["dspy", "ai", "language models", "llm", "openai"]
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.12"
-pydantic = ">=2.5.0,<=2.7"
```

Comment:
You could use `pydantic="^2.0"` which also ensures that it would be `<3.0.0`. See https://python-poetry.org/docs/dependency-specification/

Reply:
Good call. Thanks!